### ISSUE TITLE
Change out Supervisor for better init system

### ISSUE THREAD

**Issue by jaydrogers** [serversideup/docker-php#15]
# Background
* Currently the `fpm-apache` and `fpm-nginx` images use Supervisor to bring things up
* According to some reading online, supervisor is not the best for this because it could lead to zombie processes or not properly handling errors

# Possible Solution
* Use [S6 Overlay](https://github.com/just-containers/s6-overlay)
* @brianlmoon also recommended this: https://github.com/jwilder/dockerize

### More details
* This is a process supervisor specifically built for Docker
* S6 is used in many LinuxServer.io images

### Hesitations
* I might want to look into Swoole more (#14) because this could be where we just leave FPM all together (🥳)
* The only thing I would need to understand is how can Wordpress run with Apache + Swoole
* If I can get things to work, I might be able to skip S6 all together

**Comment by szepeviktor** [comment]
This is a very good idea.

**Comment by szepeviktor** [comment]
> The only thing I would need to understand is how can Wordpress run with Apache + Swoole

1. Dangerous people spell WordPress as _Wordpress_
2. WordPress has no HTTP kernel, it needs to boot up every time - no other way to run it

**Comment by jaydrogers** [comment]
> Maybe swoole can be another variant vs php-fpm?

Absolutely. I would definitely offer Swoole as it's own variant.

Thanks for the heads up on Swoole. I haven't used it before, so I will definitely looking out for this as I start to toy around with it.

I will do some playing around with S6 and keep ya'll posted! 🤠

**Comment by jaydrogers** [comment]
Goodbye Supervisor, hello S6 Overlay!

I just merged the S6 enhancements into `main`.

I spent a lot of time toying with S6 Overlay and I feel good about where it is at now. There still (and always will be) some improvements along the way, but I have been running these images in production with a few of our apps and it's been working great!

If you're interested, check it out and let me know your feedback. Thanks for chiming in guys! 🙌
